extensions/amp-ad-network-doubleclick-impl/0.1/safeframe-host.js

```diff
@@ -0,0 +1,580 @@
+                                 payload.expand_t);
+    let expandWidth = Math.floor(payload.expand_r +
+                                 payload.expand_l);
+    // We assume that a fair amount of usage of this API will try to pass the final
+    // size of the iframe that is desired, instead of the correct usage which is
+    // specify how much to expand by. We try to protect against this, by detecting
+    // if the requested expansion size matches the size of the amp-ad element, and
+    // if so we assume that the desired effect was to resize to fill the element.
+    if (expandWidth != this.baseInstance_.element.style.width.split('px')[0] ||
+       expandHeight != this.baseInstance_.element.style.height.split('px')[0]) {
+      expandWidth += Number(this.iframe_.width);
+      expandHeight += Number(this.iframe_.height);
+    }
+    this.handleSizeChange(expandHeight,
+        expandWidth,
+        SERVICE.EXPAND_RESPONSE);
+  }
+
+  /**
+   * @private
+   */
+  handleCollapseRequest_() {
+    this.handleSizeChange(this.initialSize_.height,
+        this.initialSize_.width,
+        SERVICE.COLLAPSE_RESPONSE,
+        /** isCollapse */ true);
+  }
+
+  /**
+   * @param {number} height
+   * @param {number} width
+   */
+  resizeIframe(height, width) {
+    this.iframe_.style.height = height + 'px';
+    this.iframe_.style.width = width + 'px';
+  }
+
+  /**
+   * Resizes the safeframe, and potentially the containing amp-ad element.
+   * Then sends a response message to the Safeframe creative.
+   * @param {number} height In pixels.
+   * @param {number} width In pixels.
+   * @param {string} message
+   * @param {boolean=} optIsCollapse Whether this is a collapse attempt.
+   */
+  handleSizeChange(height, width, message, optIsCollapse) {
+    // If the new size is fully contained within the bounds of the amp-ad,
+    // we can resize immediately as there will be no reflow.
+    if (!optIsCollapse &&
+        width <= this.creativeSize_.width &&
+        height <= this.creativeSize_.height) {
+      this.resizeIframe(height, width);
+      this.sendResizeResponse(/** SUCCESS */ true, message);
+    } else {
+      this.resizeAmpAdAndSafeframe(
+          height, width, message, optIsCollapse);
+    }
+  }
+
+  /**
+   * @param {boolean} success
+   * @param {string} message
+   */
+  sendResizeResponse(success, message) {
+    this.sendMessage_({
+      uid: this.uid,
+      success,
+      newGeometry: this.getCurrentGeometry(),
+      'expand_t': this.currentGeometry_.allowedExpansion_t,
+      'expand_b': this.currentGeometry_.allowedExpansion_b,
+      'expand_r': this.currentGeometry_.allowedExpansion_r,
+      'expand_l': this.currentGeometry_.allowedExpansion_l,
+      push: true,
+    }, message);
+  }
+
+  /**
+   *
+   * @param {number} height
+   * @param {number} width
+   * @param {string} message
+   * @param {boolean=} optIsCollapse
+   */
+  resizeAmpAdAndSafeframe(height, width, message, optIsCollapse) {
```

Comment:
Might need to change optIsCollapse -> opt_isCollapse.

Reply:
I think the lint rules changed, this is fine now